### PR TITLE
[FIX] web_editor: allow widget text on field html in studio

### PR DESCRIPTION
**Pull request by JKE-be**
Before this commit, widget='text' was not available in web_studio since 'html'
was missing from the supportedFieldTypes list of the TextField.

After this commit you can use widget text (and other derivated type of text).

task-sbu-feedback

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/backend/field_html.js [hunk 1]
```diff
@@ -517,6 +517,8 @@ var FieldHtml = basic_fields.DebouncedField.extend(TranslatableFieldMixin, {
 
 field_registry.add('html', FieldHtml);
 
+// allow to use widget = 'text' on html field
+field_registry.map.text.prototype.supportedFieldTypes.push('html');
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/backend/field_html.js]: I think you can just modify the line `supportedFieldTypes: ['text'],` into `supportedFieldTypes: ['text', 'html'],` in web, no? The feature does not seem related to web_editor.

Comment (JKE-be) [reply addons/web_editor/static/src/js/backend/field_html.js]: don't like the idea to modify web to add support of feature in web_editor.

If you just have web, you are not aware of this htmlfield.

Seems more correct like that.

Comment (JKE-be) [reply addons/web_editor/static/src/js/backend/field_html.js]: after the real question, is... Why fieldHtml is not in web instead of web_editor ;)

Comment (qsm-odoo) [reply addons/web_editor/static/src/js/backend/field_html.js]: You don't use web_editor here, I think you are confused with the html field *widget* :wink: Your PR is making the text field widget of *web* available for python html field (of *base*). No web_editor involved to me. Or I misunderstand the original problem.

Comment (qsm-odoo) [reply addons/web_editor/static/src/js/backend/field_html.js]: 'html' in your new line does not refer to the web_editor 'html' field but to the python 'html' field type

**Comment by JKE-be** [comment]
@robodoo r+
